src/controllers/llmo/llmo-onboarding.js

```diff
@@ -0,0 +1,373 @@
+/*
+ * Copyright 2025 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/* c8 ignore start */
+
+import { Config } from '@adobe/spacecat-shared-data-access/src/models/site/config.js';
+import { createFrom } from '@adobe/spacecat-helix-content-sdk';
+import { Octokit } from '@octokit/rest';
+import { Entitlement as EntitlementModel } from '@adobe/spacecat-shared-data-access/src/models/entitlement/index.js';
+import TierClient from '@adobe/spacecat-shared-tier-client';
+
+// LLMO Constants
+const LLMO_PRODUCT_CODE = EntitlementModel.PRODUCT_CODES.LLMO;
+const LLMO_TIER = EntitlementModel.TIERS.FREE_TRIAL;
+const SHAREPOINT_URL = 'https://adobe.sharepoint.com/:x:/r/sites/HelixProjects/Shared%20Documents/sites/elmo-ui-data';
+
+/**
+ * Generates the data folder name from a domain.
+ * @param {string} domain - The domain name
+ * @param {string} env - The environment (prod, dev, etc.)
+ * @returns {string} The data folder name
+ */
+export function generateDataFolder(domain, env = 'dev') {
+  const baseURL = domain.startsWith('http') ? domain : `https://${domain}`;
```

Comment:
please use baseURL as argument to the function, instead of domain. At this point baseURL should have been resolved from domain